utils/DMTHandler.js

```diff
@@ -1,154 +1,193 @@
+                const newRetryCount = retryCount - 1;
+                setTimeout(() => {
+                  return this.sendEventForTransformation(payload, newRetryCount)
+                    .then(resolve)
+                    .catch(reject);
+                }, 500);
+              } else {
+                // Even after all the retries event transformation
+                // is not successful, ignore the event
+                reject(`[Transformation]:: Transformation failed with status ${status}`);
+                return;
+              }
+            } catch (err) {
+              reject(err);
             }
-          } catch (err) {
-            reject(err);
           }
-        }
-      };
-
-      xhr.send(JSON.stringify(payload, replacer));
-    } catch (error) {
-      reject(error);
-    }
-  });
-};
-
-/**
- * A helper function that will process the transformation
- * and return the transformed event payload
- *
- */
-const processTransformation = (event, writeKey, dataPlaneUrl, cb) => {
-  // createPayload
-  const payload = createPayload(event);
+        };
 
-  // Send event for transformation with payload, writekey and retryCount
-  sendEventForTransformation(payload, writeKey, dataPlaneUrl, retryCount)
-    .then((outcome) => {
-      return cb(outcome);
-    })
-    .catch((err) => {
-      if (typeof err === 'string') {
-        logger.error(err);
-      } else {
-        logger.error(err.message);
+        xhr.send(JSON.stringify(payload, replacer));
+      } catch (error) {
+        reject(error);
       }
-      // send null as response in case of error or retry fail
-      return cb(null);
     });
-};
+  }
+
+  /**
+   * A helper function that will process the transformation
+   * and return the transformed event payload
+   *
+   */
+  processTransformation() {
+    this.isTransformationProcessing = true;
+    const cur = this.queue.shift();
+    const payload = this.createPayload(cur.event);
+
+    // Send event for transformation with payload, writekey and retryCount
+    this.sendEventForTransformation(payload, this.retryCount)
+      .then((outcome) => {
+        this.isTransformationProcessing = false;
+        return cur.cb(outcome);
+      })
+      .catch((err) => {
+        if (typeof err === 'string') {
+          logger.error(err);
+        } else {
+          logger.error(err.message);
+        }
+        this.isTransformationProcessing = false;
+        // send null as response in case of error or retry fail
+        return cur.cb({ transformedPayload: null });
+      });
+  }
+
+  start() {
+    if (this.isTransformationProcessing && !this.isTimerStarted) {
+      const self = this;
+      const interval = setInterval(() => {
+        this.isTimerStarted = true;
+        if (self.queue.length === 0) {
+          clearInterval(interval);
+          this.isTimerStarted = false;
+        }
+        if (!self.isTransformationProcessing) {
+          self.processTransformation();
+        }
+      }, 100);
```

Comment:
We shouldn't wait a fixed amount of time to process the next item in the queue; it should immediately begin at the end of processing the previous queue item.

Reply:
Addressed.